### ISSUE TITLE
Fails to build on arm32 Darwin, missing aes_*_internal symbols

### ISSUE THREAD

**Issue by Un1q32** [libressl/portable#1113]
crypto/arch/arm/crypto_arch.h defines the macros indicating that the platform supports assembly implementations of those functions, but those assembly files are only built on ELF platforms.  Those macros should either be behind an `#ifdef __ELF__`, or the assembly functions should be made to build for other binary formats.

**Comment by botovq** [comment]
This is with cmake, right?

If I understand correctly, the autoconf build disables assembly by passing `-DOPENSSL_NO_ASM` and this is an issue with CMake's ENABLE_ASM logic on this platform. As a workaround using `cmake -DENABLE_ASM=OFF ...` should do the trick.

**Comment by Un1q32** [comment]
Yes, passing -DENABLE_ASM=OFF works, but this should be done by default, the build shouldn't fail when it isn't passed.

**Comment by botovq** [comment]
Agreed. Hence "issue", "workaround". It will be fixed soon.

**Comment by Un1q32** [comment]
This is still an issue in 4.1.0, except worse since there are more defines that aren't guarded behind `__ELF__`
